server/src/main/java/org/apache/druid/server/coordinator/loading/HttpLoadQueuePeon.java

```diff
@@ -76,14 +76,10 @@
 public class HttpLoadQueuePeon implements LoadQueuePeon
 {
   public static final TypeReference<List<DataSegmentChangeRequest>> REQUEST_ENTITY_TYPE_REF =
-      new TypeReference<List<DataSegmentChangeRequest>>()
-      {
-      };
+      new TypeReference<>() {};
```

Comment:
Java 9+ supports diamond operator with anonymous classes.
Since we have dropped support for Java 8, we can use this syntax here (and in several other places in the code which can be done in a separate PR).

cc: @Akshat-Jain

Reply:
Have raised PR: https://github.com/apache/druid/pull/17567. The PR isn't limited to just anonymous class usages, but rather does it for all usages where explicit type argument was redundant. I don't see an easy way to identify usages only for anonymous classes, and I didn't see any reason why we shouldn't do it for other usages as well, so ended up doing that. Hope that works!

Appreciate your review on the PR, thanks!